server.go

```diff
@@ -1347,24 +1362,25 @@ func (s *Server) processUnaryRPC(ctx context.Context, t transport.ServerTranspor
 		t.IncrMsgRecv()
 	}
 	df := func(v any) error {
-		defer cancel()
-
-		if err := s.getCodec(stream.ContentSubtype()).Unmarshal(d, v); err != nil {
+		defer d.Free()
+		if err := s.getCodec(stream.ContentSubtype()).Unmarshal(d.Ref(), v); err != nil {
 			return status.Errorf(codes.Internal, "grpc: error unmarshalling request: %v", err)
 		}
-		for _, sh := range shs {
-			sh.HandleRPC(ctx, &stats.InPayload{
-				RecvTime:         time.Now(),
-				Payload:          v,
-				Length:           len(d),
-				WireLength:       payInfo.compressedLength + headerLen,
-				CompressedLength: payInfo.compressedLength,
-				Data:             d,
-			})
+
+		if len(shs) != 0 {
```

Comment:
These `len` checks only seem to increase the indent of the code inside the `for` loop.

Reply:
Same as https://github.com/PapaCharlie/grpc-go/pull/1#discussion_r1655395873, this used to be necessary, now it's not